diff --git a/node_modules/jose/dist/node/cjs/runtime/check_modulus_length.js b/node_modules/jose/dist/node/cjs/runtime/check_modulus_length.js
index 72c670e..cec7685 100644
--- a/node_modules/jose/dist/node/cjs/runtime/check_modulus_length.js
+++ b/node_modules/jose/dist/node/cjs/runtime/check_modulus_length.js
@@ -46,7 +46,7 @@ const setModulusLength = (keyObject, modulusLength) => {
 };
 exports.setModulusLength = setModulusLength;
 exports.default = (key, alg) => {
-    if (getModulusLength(key) < 2048) {
+    if (getModulusLength(key) < 1024) {
         throw new TypeError(`${alg} requires key modulusLength to be 2048 bits or larger`);
     }
 };
